src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpMessageOptions.cs

```diff
@@ -5,49 +5,49 @@
 
 namespace Azure.Base.Http
 {
-    public abstract partial class HttpMessage
+    public partial class HttpPipelineContext
```

Comment:
@KrzysztofCwalina do you have strong feelings about `HttpMessageOptions`? I was thinking about turning it into `object Get(string)` and `void Set(string, object)` methods on context/message.

Reply:
I think the main scenario for this is to pass System.Type as the key. For example, RetryPolicy would try to access its strongly typed options by calling message.Options.Get(typeof<RetryPolicy>). This is great to avoid conflicts and also if the Type is internal, it can prevent (or make it surer hard) other code from reading it.